### ISSUE TITLE
Angular CLI with Kendo Combobox: Build fails with build-optimizer flag turned on

### ISSUE THREAD

**Issue by prabh-62** [telerik/kendo-angular#846]
## I'm submitting a...
* Bug report 

## Current behavior
- Create a new project with [Angular CLI](https://cli.angular.io/) 
- Import Angular Kendo Combobox library
- Enable the build-optimizer flag in Angular CLI to optimize the final output dist files
- Build script errors out with kendo Combobox Module

## Expected behavior
- Build script should not produce any errors with Kendo Combobox Module

## Minimal reproduction of the problem with instructions
```
> ng new kendo-drop --minimal
> npm install @progress/kendo-angular-{dropdowns,l10n}
```
In package.json, amend the build script
```
"build": "ng build --target production --aot true --build-optimizer true --stats-json true"
```

In app.module.ts, replace code with this snippet
```typescript
@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    DropDownsModule,
    FormsModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
```

In app.component.ts, add this code (Copied from [official docs](http://www.telerik.com/kendo-angular-ui/components/dropdowns/combobox/#toc-basic-usage))
```typescript
@Component({
  selector: 'my-app',
  template: `
    <div class="example-config">
      <input id="ac" type="checkbox" [(ngModel)]="allowCustom">
      <label for="ac">Allow custom values</label>
    </div>
    <div class="example-wrapper">
    <p>Favorite sport:</p>
    <kendo-combobox [data]="listItems" [allowCustom]="allowCustom">
    </kendo-combobox>
    </div>
  `
})
class AppComponent {
    public allowCustom: boolean = true;
    public listItems: Array<string> = ["Baseball", "Basketball", "Cricket", "Field Hockey", "Football", "Table Tennis", "Tennis", "Volleyball"];
}
```

Now run the build script 
```
> npm run build
```

```
ERROR in ./node_modules/rxjs/observable/BoundCallbackObservable.js
Module build failed: TypeError: Cannot read property 'type' of undefined
```

## What is the motivation or use case for changing the behavior?
Build-optimizer reduces the size of final output files by a large margin and I would like to keep it turned on

## Environment

Package versions:
```
npm ls --depth 0

├── @angular/animations@4.3.5
├── @angular/cli@1.3.1
├── @angular/common@4.3.5
├── @angular/compiler@4.3.5
├── @angular/compiler-cli@4.3.5
├── @angular/core@4.3.5
├── @angular/forms@4.3.5
├── @angular/http@4.3.5
├── @angular/language-service@4.3.5
├── @angular/platform-browser@4.3.5
├── @angular/platform-browser-dynamic@4.3.5
├── @angular/router@4.3.5
├── @progress/kendo-angular-dropdowns@1.1.3
├── @progress/kendo-angular-l10n@1.0.3
├── core-js@2.5.0
├── rxjs@5.4.3
├── typescript@2.4.2
└── zone.js@0.8.16
```

Browser:
- Chrome (desktop) version 62.0.3192.0

System:
- TypeScript version: 2.4.2
- Node version: v8.4.0 (NPM v5.3.0)
- Angular CLI: v1.3.1
- Platform:  Ubuntu 16.04.3 LTS

**Comment by tsvetomir** [comment]
The only thing that seems to help is downgrading Typescript to 2.4.0:

```
npm install --save-dev typescript@2.4.0
```

The problem seems to be caused by a code fragment bundled in rxjs. I'll try to extract a minimal reproduction sample out of and file with the [Typescript project](https://github.com/Microsoft/TypeScript).

**Comment by tsvetomir** [comment]
Logged the issue as Microsoft/TypeScript#18011

**Comment by tsvetomir** [comment]
Should be fixed in Typescript v2.5.1

**Comment by prabh-62** [comment]
Thank you for investigating @tsvetomir!
